pyvista/examples/downloads.py

```diff
@@ -3468,12 +3468,18 @@ def download_single_sphere_animation(load=True):  # pragma: no cover
 
     Examples
     --------
+    >>> import os
+    >>> import tempfile
     >>> import pyvista
     >>> from pyvista import examples
     >>> filename = examples.download_single_sphere_animation(load=False)
     >>> reader = pyvista.PVDReader(filename)
     >>> plotter = pyvista.Plotter()
-    >>> plotter.open_gif("single_sphere_pvd.gif")
+    >>> filename = os.path.join(
+    ...     tempfile._get_default_tempdir(),
```

Comment:
I cannot easily find documentation on what these private methods do.  I assume those files are stored in temporary storage, but not explicitly cleaned up by the python process.  A private method would be confusing in a documentation example IMO.

Is there a need to use this over `TemporaryFile` or `TemporaryDirectory`?  I believe the latter can be used as a context manager to delete the directory and files inside when done.

Either way: since this is an example that documentation readers may be copying and pasting and running directly, we should also be wary of adding too much special magic.  They may not be able to find the file.  One thought is to put in a commented line about that the users can uncomment, but this would get tricky with the context manager approach.

I'm not really sure what is the best approach.

Reply:
I agree about not using private methods in documentation (I'd even argue against relying them in PyVista proper).

There's [a very similar example for `numpy.memmap`](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html), we could do something along the same lines (including pointing out explicitly that normally one would use a real filename).

Reply:
Fixed in d02372d82d07c4e0fdd3dac069c06b926f096d2c.